toolchain/semantics/semantics_node_kind.def

```diff
@@ -15,6 +15,9 @@
 
 CARBON_SEMANTICS_NODE_KIND(Invalid)
 
+// A cross-reference between IRs.
+CARBON_SEMANTICS_NODE_KIND(CrossReference)
```

Comment:
For a follow-up PR: what do you think about just calling these *references*? I don't think there is anything "cross" any more about them, and that seems fine?

(Clang for example calls analogous nodes in its AST "DeclRef"s.)

Reply:
I'm wary of that, I suppose. To me, all the index nodes are references, because they're within the same IR. A reference crosses IR boundaries; to me, that's the cross-reference: it's not the same IR, it's another IR.

Reply:
Oh, that's a different meaning of the "cross" than I had guessed. Let's maybe brainstorm a bit in Discord or elsewhere, see if there is a name that will more reliably capture this aspect. Regardless, not for this PR. =D

Reply:
That is, I've been considering renaming everything Id -> Ref because "Id" has been feeling odd.